### PR TITLE
DOC: how to_datetime %S differs from strptime

### LINKED ISSUE #47134: DOC: how to_datetime %S differs from strptime
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.to_datetime.html?highlight=to_datetime#pandas.to_datetime

### Documentation problem

The documentation says that the formatting strings used by `pandas.to_datetime` are the same as `datetime.strptime`, other than the number of digits `%f` handles.

This is not quite true, there is another difference. `pandas.to_datetime` with `%S` (without `%f`) will capture microseconds. `strptime` will throw an error. (Which makes pandas.to_datetime more useful than strptime)

```
import pandas as pd
import datetime as dt
fmt = "%Y/%m/%d %H:%M:%S"
s= "2021/01/01 12:01:01.123"
assert pd.to_datetime(s, format=fmt).microsecond > 0
dt.datetime.strptime(s, fmt)
```

The second-last line passes,
the last line throws `ValueError: unconverted data remains: .123`.

### Suggested fix for documentation

Change:

> Note that "%f" will parse all the way up to nanoseconds.

to:

> Note that "%S" without "%f" will capture fractional seconds up to nanoseconds,
> Note that "%f" will parse all the way up to nanoseconds.
> "%S" without "%f" will capture all the way up to nanoseconds if present as decimal places, but will also handle the case where the number of seconds is an integer.

### PR DESCRIPTION
**Pull request by yadomkar**
- [ ] closes #47134
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by heroesIV** [comment]
This is my first PR. Do let me know if any changes are needed. Open to all feedback.

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -777,11 +777,18 @@ def to_datetime(
...
         <https://docs.python.org/3/library/datetime.html
         #strftime-and-strptime-behavior>`_ for more information on choices.
+
+        - Note that :const:`"%f"` will parse all the way
+          up to nanoseconds.
+
+        - Note that :const:`"%S"` without :const:`"%f"` will capture all the way
+          up to nanoseconds if present as decimal places, but will also handle
+          the case where the number of seconds is an integer.
+
```
Comment (MarcoGorelli) [inline pandas/core/tools/datetimes.py:791]:
```suggestion
        #strftime-and-strptime-behavior>`_ for more information on choices, though
        note the following differences:

        - :const:`"%f"` will parse all the way
          up to nanoseconds;

        - :const:`"%S"` without :const:`"%f"` will capture all the way
          up to nanoseconds if present as decimal places, and will also handle
          the case where the number of seconds is an integer.

```

File: pandas/core/tools/datetimes.py [hunk 2]
```diff
@@ -950,6 +957,21 @@ def to_datetime(
     DatetimeIndex(['1960-01-02', '1960-01-03', '1960-01-04'],
                   dtype='datetime64[ns]', freq=None)
 
+    **Exceptions to strptime behavior**
```
Comment (MarcoGorelli) [inline pandas/core/tools/datetimes.py:960]:
```suggestion
    **Differences with strptime behavior**
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @heroesIV , just suggesting a slight rewording

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me

**Comment by gfyoung** [comment]
Thanks @heroesIV !

**Pull request merged**